ca_gazebo/src/create_virtual_wall_plugin.cpp

```diff
@@ -0,0 +1,147 @@
+#include <cmath>
+
+#include <gazebo/gazebo.hh>
+#include <gazebo/physics/physics.hh>
+#include <gazebo/common/common.hh>
+#include <gazebo/sensors/RaySensor.hh>
+#include <ignition/math/Vector3.hh>
+#include <ignition/math/Pose3.hh>
+#include <ros/ros.h>
+#include <std_msgs/Bool.h>
+
+#include <ca_gazebo/create_virtual_wall_plugin.hh>
+
+static const char create2_model_name_prefix[] = "irobot_create2.";
+static const size_t create2_model_name_prefix_length = sizeof(create2_model_name_prefix) - 1;
+
+using gazebo::VirtualWallSensorPlugin;
+
+GZ_REGISTER_SENSOR_PLUGIN(VirtualWallSensorPlugin)
+
+void VirtualWallSensorPlugin::Load(gazebo::sensors::SensorPtr sensor, sdf::ElementPtr sdf)
+{
+  this->sensor_ = std::dynamic_pointer_cast<sensors::RaySensor>(sensor);
+  this->world_ = physics::get_world();
+  this->sensor_->SetActive(true);
+  ROS_DEBUG_NAMED("virtual_wall_plugin", "Loading");
+
+  if (!ros::isInitialized()) {
+      int argc = 0;
+      char** argv = NULL;
+      ros::init(argc,argv,"bumper_node",ros::init_options::AnonymousName);
+  }
+  this->rosnode_ = std::make_unique<ros::NodeHandle>();
+  this->updateNewEntity_ = event::Events::ConnectAddEntity(
+    std::bind(&VirtualWallSensorPlugin::OnAddEntity, this));
+  this->OnAddEntity();
+  // Listen to the update event. This event is broadcast every
+  // simulation iteration.
+  this->updateConnection_ = event::Events::ConnectWorldUpdateBegin(
+      std::bind(&VirtualWallSensorPlugin::OnUpdate, this));
+  ROS_DEBUG_NAMED("virtual_wall_plugin", "Loaded");
+}
+
+void VirtualWallSensorPlugin::OnAddEntity()
+{
+  const auto & world_models = this->world_->Models();
+  // Filter models which name starts with `irobot_create2.` and are continued with a number.
+  create_2_models_.resize(0);
+  pubs_.resize(0);
+  std::copy_if(
+    world_models.begin(),
+    world_models.end(),
+    std::back_inserter(this->create_2_models_),
+    [this] (const physics::ModelPtr & model)
+    {
+      const std::string & name = model->GetName();
+      ROS_DEBUG_NAMED("virtual_wall_plugin", "checking model %s", name.c_str());
+      if (name.size() < create2_model_name_prefix_length + 1) {
+        ROS_DEBUG_NAMED("virtual_wall_plugin", "model name is short");
```

Comment:
One thing that applies for all `ROS_DEBUG_NAMED()` in this method. Do something like this:

```c++
ROS_DEBUG_STREAM_NAMED("virtual_wall_plugin", "model name " << name << "...");
```

So, print also the model name that is causing that.